knowledge_graph.py

```diff
@@ -0,0 +1,110 @@
+"""Knowledge Graph"""
+from typing import Dict, List, Any
+from transformers import (
+    Text2TextGenerationPipeline,
+    AutoModelForSeq2SeqLM,
+    AutoTokenizer)
+from transformers.pipelines.text2text_generation import ReturnType
+from pydantic import Field
+
+class KnowledgeGraph():
+  """Knowledge Graph class"""
+
+  max_new_tokens: int = Field(
+    default=200, description="Maximum number of new tokens to generate."
+  )
+  predictions : List[Dict[str, Any]] = Field(default=[])
+
+  def __init__(
+    self,
+    *args,
+    model_name:str='Babelscape/rebel-large',
+    **kwargs):
+
+    self.model = EntityRelationshipExtractionPipeline(model=model_name, *args, **kwargs)
+
+  def inference(self, sentence: str) -> Dict[str, Any]:
+    """inference"""
+    self.predictions = self.model(sentence)
+
+    return self.predictions
+
+class EntityRelationshipExtractionPipeline(Text2TextGenerationPipeline):
+  """Key phrase extractor pipeline class"""
+  def __init__(self, model, *args, **kwargs):
+    super().__init__(
+      model=AutoModelForSeq2SeqLM.from_pretrained(model),
+      tokenizer=AutoTokenizer.from_pretrained(model),
+      *args,
+      **kwargs)
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `EntityRelationshipExtractionPipeline` class is missing a docstring for the `__init__` method. Docstrings are crucial for understanding the purpose and usage of methods.

```diff
  def __init__(self, model, *args, **kwargs):
+    """
+    Initializes the EntityRelationshipExtractionPipeline with a specific model.
+    
+    Args:
+        model (str): The name of the model to use for text generation.
+        *args: Variable length argument list.
+        **kwargs: Arbitrary keyword arguments.
+    """
    super().__init__(
      model=AutoModelForSeq2SeqLM.from_pretrained(model),
      tokenizer=AutoTokenizer.from_pretrained(model),
      *args,
      **kwargs)
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->